crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -896,11 +934,29 @@ impl Object {
                     checkpoint_sequence_number: history_object.checkpoint_sequence_number,
                 }),
                 checkpoint_viewed_at,
+                root_version: None,
             }),
         }
     }
 }
 
+/// We're deliberately choosing to use a child object's version as the root here, and letting the
+/// caller override it with the actual root object's version if it has access to it.
+///
+/// Using the child object's version as the root means that we're seeing the dynamic field tree
+/// under this object at the state resulting from the transaction that produced this version.
```

Comment:
This might make more sense as a doc comment for `fn value` of `DynamicField`, something like "If the `value` is a child object, its contents come from the latest version that is at most equal to its parent object's version."

Reply:
If that makes sense, you'll need to run `snapshot_tests.rs` to regenerate the schema and update the snapshot test

Reply:
This is what I came up for `DynamicField.value`:
```
     /// The actual data stored in the dynamic field.
     /// The returned dynamic field is an object if its return type is MoveObject,
-    /// in which case it is also accessible off-chain via its address.
+    /// in which case it is also accessible off-chain via its address and the latest
+    /// version that is at most equal to its parent object's version.
```
Good to go?

Reply:
Apologies in advance for bikeshedding 😅 but it's possible for someone to access the child object directly off-chain via its address and at any version, so this is not entirely accurate. Proposing...

The returned dynamic field is an object if its return type is `MoveObject`, and its contents will be from the latest version that is at most equal to its parent object's version. Otherwise, it is also accessible off-chain via its address.